src/track/taglib/trackmetadata_id3v2.cpp

```diff
@@ -83,24 +83,45 @@ inline QString frameToQString(
     return toQString(frame.toString());
 }
 
+bool logWarningIfUnsupportedID3v2Frame(const TagLib::ID3v2::Frame* pFrame) {
+    if (dynamic_cast<const TagLib::ID3v2::UnknownFrame*>(pFrame)) {
+        kLogger.warning()
+                << "ID3v2 frame"
+                << pFrame->frameID().data()
+                << "is not yet supported by TagLib";
+        return true;
+    }
+    // Not an of UnknownFrame, i.e. maybe some other kind of ID3v2 frame
+    return false;
+}
+
+void logWarningAboutUnsupportedOrUnexpectedID3v2Frame(const TagLib::ID3v2::Frame* pFrame) {
+    if (!logWarningIfUnsupportedID3v2Frame(pFrame)) {
+        // Do not crash if the caller unexpectedly passed a nullptr
+        VERIFY_OR_DEBUG_ASSERT(pFrame) {
+            return;
+        }
+        kLogger.warning()
+                << "Unexpected ID3v2 frame"
+                << pFrame->frameID().data();
+    }
+}
+
 // Returns the first frame of an ID3v2 tag as a string.
 QString firstNonEmptyFrameToQString(
         const TagLib::ID3v2::FrameList& frameList) {
     for (const TagLib::ID3v2::Frame* pFrame : frameList) {
-        if (pFrame) {
-            TagLib::String str = pFrame->toString();
-            if (!str.isEmpty()) {
-                return toQString(str);
-            }
-            const auto* pUnknownFrame = dynamic_cast<const TagLib::ID3v2::UnknownFrame*>(pFrame);
-            if (pUnknownFrame) {
-                kLogger.warning()
-                        << "Unsupported ID3v2 frame"
-                        << pUnknownFrame->frameID().data();
-            }
+        VERIFY_OR_DEBUG_ASSERT(pFrame) {
```

Comment:
```suggestion
        VERIFY_OR_DEBUG_ASSERT(!pFrame) {
```
backwards again.

Reply:
This is not how `VERIFY_OR_DEBUG_ASSERT` works. Either the expression evaluates to "true", or we throw a debug assertion and run the code within the curly braces.

Reply:
ah right, I'm sorry.